pkg/registry/componentstatus/rest.go

```diff
@@ -53,11 +54,22 @@ func (rs *REST) NewList() runtime.Object {
 func (rs *REST) List(ctx api.Context, options *unversioned.ListOptions) (runtime.Object, error) {
 	servers := rs.GetServersToValidate()
 
-	// TODO: This should be parallelized.
-	reply := []api.ComponentStatus{}
+	wait := sync.WaitGroup{}
+	wait.Add(len(servers))
+	statuses := make(chan api.ComponentStatus, len(servers))
 	for name, server := range servers {
-		status := rs.getComponentStatus(name, server)
-		reply = append(reply, *status)
+		go func() {
+			defer wait.Done()
+			status := rs.getComponentStatus(name, server)
```

Comment:
This won't work. Due to Go's scoping rules for the "for loop" vars, the go function will not evaluate "name" and "server" in the way you want (I believe it'll only get the last entry). See https://play.golang.org/p/yRbY3uRb39 for an example.